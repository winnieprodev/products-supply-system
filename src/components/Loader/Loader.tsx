import LogoImg from "assets/images/logo.svg";
import React from "react";
import { Empty } from "utils/types";

import * as Styled from "./loader.styled";

export const Loader: React.FC<Empty> = () => (
  <Styled.Circle>
    <Styled.LogoImage src={LogoImg} />
    <Styled.LogoImage $delay={0.4} src={LogoImg} />
    <Styled.LogoImage $delay={0.7} src={LogoImg} />
  </Styled.Circle>
);

export default React.memo(Loader);
