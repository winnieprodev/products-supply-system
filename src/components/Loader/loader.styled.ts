import styled, { css, keyframes } from "styled-components";
import { secondaryColorLighter } from "styles/colors";

const CirclePulse = keyframes`
    0% { transform: scale(0.4) }
    10% { transform: scale(0.7) }
    20% { transform: scale(0.5) }
    30% { transform: scale(0.75) }
    40% { transform: scale(0.6) }
    50% { transform: scale(0.85) }
    60% { transform: scale(0.7) }
    70% { transform: scale(1) }
    80% { transform: scale(0.8) }
    90% { transform: scale(1.1) }
    100% { transform: scale(0.4) }
`;

export const Circle = styled.div`
  width: 48px;
  height: 48px;
  padding: 8px;
  position: relative;

  &:before {
    position: absolute;
    right: 0;
    top: 0;
    left: 0;
    bottom: 0;
    z-index: -1;
    background: ${secondaryColorLighter};
    display: block;
    content: "";
    border-radius: 50%;
    animation: ${CirclePulse} 5s infinite;
  }
`;

const LogoRotate = keyframes`
    0% { transform: rotate(0deg) }
    10% { transform: rotate(0deg) }
    20% { transform: rotate(180deg)}
    80% { transform: rotate(-720deg)}
    100% { transform: rotate(0)}
`;

interface LogoImageProps {
  $delay?: number;
}
export const LogoImage = styled.img<LogoImageProps>`
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  animation: ${LogoRotate} 10s infinite;
  animation-delay: ${({ $delay = 0 }) => $delay}s;
  ${({ $delay }) =>
    $delay &&
    css`
      opacity: 0.3;
    `}
`;
