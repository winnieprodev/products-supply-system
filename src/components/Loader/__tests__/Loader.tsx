import { render } from "@testing-library/react";

import Loader from "../Loader";

describe("Loader", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<Loader />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
