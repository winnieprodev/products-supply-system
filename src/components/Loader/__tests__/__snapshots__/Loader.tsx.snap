// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Snapshot tests it matches the snapshot 1`] = `
.c0 {
  width: 48px;
  height: 48px;
  padding: 8px;
  position: relative;
}

.c0:before {
  position: absolute;
  right: 0;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: -1;
  background: #F3F3F3;
  display: block;
  content: "";
  border-radius: 50%;
  -webkit-animation: hAYQBa 5s infinite;
  animation: hAYQBa 5s infinite;
}

.c1 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c2 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
  opacity: 0.3;
}

.c3 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.7s;
  animation-delay: 0.7s;
  opacity: 0.3;
}

<div
  class="c0"
>
  <img
    class="c1"
    src="logo.svg"
  />
  <img
    class="c2"
    src="logo.svg"
  />
  <img
    class="c3"
    src="logo.svg"
  />
</div>
`;
